core/api/src/jsonrpc/types.rs

```diff
@@ -36,6 +37,49 @@ impl RichTransactionOrHash {
         }
     }
 }
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct Web3TransactionReceipt {
+    pub block_number:        u64,
+    pub block_hash:          H256,
+    pub contract_address:    Option<H160>,
+    pub cumulative_gas_used: U256,
+    pub effective_gas_price: U256,
+    pub from:                H160,
+    pub gas_used:            U256,
+    pub logs:                Vec<Log>,
+    pub logs_bloom:          Bloom,
+    pub status:              U256,
+    pub to:                  Option<H160>,
+    pub transaction_hash:    Hash,
+    pub transaction_index:   Option<U256>,
+    #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
+    pub transaction_type:    Option<U64>,
+}
+impl Web3TransactionReceipt {
+    pub fn from(receipt: Receipt, tx: SignedTransaction) -> Web3TransactionReceipt {
+        Web3TransactionReceipt {
+            block_number:        receipt.block_number,
+            block_hash:          receipt.block_hash,
+            contract_address:    None,
```

Comment:
Why is None here?

Reply:
If it is a contract transaction, then this address points to the new contract address. If it is an ordinary transaction, this field can be blank. At present, the underlying receipt does not return a constraint address, so it is temporarily replaced by none